### PR TITLE
Add lowerings for MLProgram to Util dialect for global variables

### PR DESCRIPTION
**Pull request by rsuderman**
MLProgram has stateful variables which represents stateful behavior.
CL includes lit tests along with a simple execution validating global variables,
loading, and storing values.

### CODE REVIEW

**Review by stellaraccident — APPROVED** [review]
Thank you for this! We will also want to convert ml_program.func but can add that in a followup.

**Comment by jpienaar** [comment]
Thanks Rob! 

A step towards #8817

File: compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      : ConversionPattern(converter, srcName, benefit, context),
+        targetName(targetName) {}
+  LogicalResult matchAndRewrite(
+      Operation *srcOp, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    SmallVector<Type> resultTypes;
+    if (failed(typeConverter->convertTypes(srcOp->getResultTypes(),
+                                           resultTypes))) {
+      return srcOp->emitError()
+             << "could not convert result types to IREE internal types";
```
Comment (jpienaar) [inline compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp:52]: In failure case would op be printed here or should we print out all the types here? (looking at implementation of convertTypes the failed type would be `srcOp->getResultTypes()[resultTypes.size()]` but that's not documented as something we can rely on, I'll ask]

File: compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  RewritePatternSet patterns(&getContext());
+  ConversionTarget target(getContext());
+  target.addLegalDialect<IREE::Util::UtilDialect>();
+  target.addIllegalDialect<ml_program::MLProgramDialect>();
+  target.markUnknownOpDynamicallyLegal([](mlir::Operation *) { return true; });
+
+  IREETypeConverter typeConverter;
+  patterns.insert<MLProgramGlobalOpPattern>(typeConverter, &getContext(), 0);
+
+  PatternBenefit specific_benefit = 100;
```
Comment (jpienaar) [inline compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp:99]: Is this just out of caution or any reason? 1 seems to be sufficient here as there are no overlapping patterns.

File: compiler/src/iree/compiler/InputConversion/Common/Passes.td [hunk 3]
```diff
@@ -15,6 +15,12 @@ def IREEImportPublic :
   let constructor = "mlir::iree_compiler::createIREEImportPublicPass()";
 }
 
+def ImportMLProgram :
+    Pass<"iree-import-ml-program", "ModuleOp"> {
```
Comment (jpienaar) [inline compiler/src/iree/compiler/InputConversion/Common/Passes.td:19]: (not for this change, but future) From the naming in the file is difficult to identify the naming convention :)

File: tests/e2e/regression/globals_ml_program.mlir [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// RUN: iree-run-mlir --iree-input-type=mhlo --iree-hal-target-backends=vmvx %s | FileCheck %s
+// RUN: [[ $IREE_VULKAN_DISABLE == 1 ]] || (iree-run-mlir --iree-input-type=mhlo --iree-hal-target-backends=vulkan-spirv %s | FileCheck %s)
```
Comment (jpienaar) [inline tests/e2e/regression/globals_ml_program.mlir:2]: OOC why do we need to test on 2 backends here?

**Pull request merged**